### PR TITLE
Miscellaneous improvements to -help messages

### PR DESCRIPTION
**Pull request by owenv**
- Add "See also" messages from the c++ driver on master
- Hide options not accepted by the current driver kind
- ~~update isAccepted(by:) to reject no-driver options in batch & interactive mode~~

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by owenv** [comment]
Oops, I missed that failure somehow when running tests locally. It looks like `-emit-module-doc-path` is listed as a no-driver option right now. I might need to change that in the swift repo and then rerun `makeOptions` before this can land.

**Comment by owenv** [comment]
I'm dropping the `isAccepted(by:)` change for now, it looks like it might require a larger refactor. This just includes the first two changes now and is passing tests for me locally.

**Comment by harlanhaskins** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Options/OptionTable.swift [hunk 1]
```diff
@@ -26,18 +26,19 @@ public struct OptionTable {
...
-      USAGE: \(usage)
+      USAGE: \(driverKind.usage)
 
       OPTIONS:
       """)
 
     for option in options {
       if option.isAlias { continue }
       if option.isHelpHidden && !includeHidden { continue }
+      guard option.isAccepted(by: driverKind) else { continue }
```
Comment (harlanhaskins) [inline Sources/SwiftDriver/Options/OptionTable.swift]: Is this still meant to be in?

Comment (owenv) [reply Sources/SwiftDriver/Options/OptionTable.swift]: Yeah, I think this is ok. As far as I can tell the current version of the acceptance check only fails to reject some invalid options, it doesn’t reject any valid ones.

**Comment by DougGregor** [comment]
@swift-ci please test

**Pull request merged**